docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1,27 +1,1777 @@
+		},
+	};
+
+	// Prepare pagination limit (how many items to return)
+	if (Object.hasOwn(body, 'limit')) {
+		const requestLimit = Number(body.limit);
+
+		if (isNaN(requestLimit)) throw new Error('Invalid limit value');
+
+		pagination.take = requestLimit > CONFIG.DEFAULT_LIMIT ? CONFIG.DEFAULT_LIMIT : requestLimit;
+	} else {
+		pagination.take = CONFIG.DEFAULT_LIMIT;
+	}
+
+	// Prepare cursor pagination (which page to return)
+	if (Object.hasOwn(body, 'cursor')) {
+		const cursor = Number(body.cursor);
+		if (isNaN(cursor)) throw new Error('Invalid cursor');
+		pagination.skip = 1;
+		pagination.cursor = {
+			id: cursor,
+		};
+	}
+
+	return pagination;
+};
+```
+
+#### API endpoints
+
+All the endpoints are defined in `server.ts`, particularly, there are two endpoints:
+
+- `/locked` to query `Locked` objects.
+- `/escrows` to query `Escrow` objects.
+
+The implementation for both endpoints is pretty straightforward. You define a list of valid query keys, such as `deleted`, `creator`, `keyId`, and `objectId` for `Locked` data and `cancelled`, `swapped`, `recipient`, and `sender` for `Escrow` data. Pass the URL query string into the pre-defined utilities to output the correct parameters that Prisma can use.
+
+```ts title='server.ts'
+import { prisma } from './db';
+import {
+	formatPaginatedResponse,
+	parsePaginationForQuery,
+	parseWhereStatement,
+	WhereParam,
+	WhereParamTypes,
+} from './utils/api-queries';
+
+app.get('/locked', async (req, res) => {
+	const acceptedQueries: WhereParam[] = [
+		{
+			key: 'deleted',
+			type: WhereParamTypes.BOOLEAN,
+		},
+		{
+			key: 'creator',
+			type: WhereParamTypes.STRING,
+		},
+		{
+			key: 'keyId',
+			type: WhereParamTypes.STRING,
+		},
+		{
+			key: 'objectId',
+			type: WhereParamTypes.STRING,
+		},
+	];
+
+	try {
+		const locked = await prisma.locked.findMany({
+			where: parseWhereStatement(req.query, acceptedQueries)!,
+			...parsePaginationForQuery(req.query),
+		});
+
+		return res.send(formatPaginatedResponse(locked));
+	} catch (e) {
+		console.error(e);
+		return res.status(400).send(e);
+	}
+});
+
+app.get('/escrows', async (req, res) => {
+	const acceptedQueries: WhereParam[] = [
+		{
+			key: 'cancelled',
+			type: WhereParamTypes.BOOLEAN,
+		},
+		{
+			key: 'swapped',
+			type: WhereParamTypes.BOOLEAN,
+		},
+		{
+			key: 'recipient',
+			type: WhereParamTypes.STRING,
+		},
+		{
+			key: 'sender',
+			type: WhereParamTypes.STRING,
+		},
+	];
+
+	try {
+		const escrows = await prisma.escrow.findMany({
+			where: parseWhereStatement(req.query, acceptedQueries)!,
+			...parsePaginationForQuery(req.query),
+		});
+
+		return res.send(formatPaginatedResponse(escrows));
+	} catch (e) {
+		console.error(e);
+		return res.status(400).send(e);
+	}
+});
+```
+
+### Next steps
+
+With the code successfully deployed on Testnet, you can now [create a frontend](#frontend) to display the trading data and to allow users to interact with the Move modules.
+
+## Frontend
```

Comment:
This section is starting to read a little like annotated code, rather than a guide, again. It's describing what every part of the finished codebase does, rather than how you build up the functionality to get to the end state.

For example, the first section talks about setting up providers -- the impression I would get from reading  that is that I should copy paste that code in, but in fact most of that code comes from running `pnpm create @mysten/dapp --template react-client-dapp`, along with setting up other things (like a `package.json`, etc) which we also haven't mentioned here.

I suspect that this is inevitable if you only have access to the final artifact and not the person who went through the journey of building it. I notice that in my own wording for how the indexer part works as well (above), where I'm not aware of the commands we might use to include prisma as a dependency, or tweak its schemas, but I can understand the output.

TL;DR we should involve @manolisliolios here and get him to explain the broad strokes of the journey.

Reply:
I was thinking the latter -- meeting with @wriches to explain more of the journey, i.e.

- Steps taken (the important thing here is to call out the order in which you did things, and when there's something that was generated, rather than written, what command was used to do that, etc).
- Non-obvious decisions and why you made them (particularly if it avoids a pitfall).